src/routes/auth/data.ts

```diff
@@ -1,19 +1,43 @@
 import { DBTables } from '../../constants/db.ts';
-import type { Profile, SignInData } from './validate.ts';
+import type { Access } from '../../types/data/access.ts';
+import type { Profile } from './validate.ts';
 
-export async function getPassword(database: D1Database, body: SignInData): Promise<string> {
+export async function getPassword(database: D1Database, email: string) {
 	const { results } = await database
-		.prepare(`SELECT password FROM ${DBTables.PROFILE} WHERE email = ? AND activatedAt IS NOT NULL LIMIT 1`)
-		.bind(body.email)
+		.prepare(`
+            SELECT password
+            FROM ${DBTables.ACCESS}
+            WHERE email = ? AND activatedAt IS NOT NULL
+			LIMIT 1
+        `)
+		.bind(email)
+		.run<string>();
+
+	const password = results[0];
+
+	return password;
+}
+
+export async function getProfileId(database: D1Database, email: string) {
+	const { results } = await database
+		.prepare(`SELECT id FROM ${DBTables.PROFILE} WHERE email = ? AND activatedAt IS NOT NULL LIMIT 1`)
+		.bind(email)
 		.run<Profile>();
 
-	const password = results[0]?.password;
+	const id = results[0]?.id;
 
-	if (!password) {
-		throw new Error('No active account found with the provided email');
-	}
+	return id;
+}
 
-	return password;
+export async function getAccessLevel(database: D1Database, profileId: string) {
+	const { results } = await database
+		.prepare(`SELECT access_level FROM ${DBTables.ACCESS} WHERE id = ? LIMIT 1`)
+		.bind(profileId)
+		.run<Access>();
```

Comment:
Double check here as well if the result is not actually an object with the `access_level` property